test/support/test_utils.ex

```diff
@@ -387,4 +387,28 @@ defmodule Logflare.TestUtils do
   end
 
   def random_pos_integer(limit \\ 1000), do: :rand.uniform(limit)
+
+  @doc """
+  Fetches data until the asserts is able to run its asserts a certain amount of times (default 10 times).
+  """
+  @spec retry_fetch(
```

Comment:
a nicer implementation that is also common in nodejs testing is to have a `waitFor` that accepts a callback, and the callback just keeps executing until it exceeds a timeout.

The interface for the function would be much simpler.

```
def assert_until(callback, opts) do
  opts = Enum.into(opts, %{start: DateTime.utc_now()} )

  //check if exceed timeout, break and throw error if exceeding 500ms
  ....

  try
    callback.()
  rescue
  _ -> 
    // recursively retry until break.
    assert_until(callback, opts)

end
```

Reply:
The caller of this function then has a nicer interface:
```
assert_until fn -> 
  assert render(view) =~ "values"
end
```

Reply:
the current usage interface of the `retry_fetch` is quite complex and error prone i think.

Reply:
just wanted something "quick and dirty" honestly and we can refactor this in a later PR as it's just for tests and reduce flakyness.